File: frontend/src/types/compassEvaluations.ts
export type EvaluationStatus = "pending" | "completed" | "failed";
export type Recommendation = "apply" | "monitor" | "skip" | "needs_review";
export type ConfidenceLevel = "low" | "medium" | "high";

export interface EvaluationSection {
  status?: string;
  score?: number | string | null;
  summary?: string | null;
  notes?: string | null;
  evidence?: string[];
  gaps?: string[];
  assumptions?: string[];
  matched_keywords?: string[];
  missing_keywords?: string[];
  [key: string]: unknown;
}

export interface EvidenceItem {
  label?: string;
  code?: string;
  message?: string;
  notes?: string | null;
  evidence?: string | string[] | null;
  status?: string;
  dimension?: string;
  severity?: string;
  score?: number;
  keywords?: string[];
}

export interface CompassEvaluation {
  id: string;
  user_id?: string;
  role_id?: string;
  workspaceId?: string;
  opportunityId?: string;
  evaluation_status?: EvaluationStatus;
  status?: EvaluationStatus | "superseded";
  overall_score?: string | null;
  overallScore?: number | null;
  recommendation?: Recommendation | string | null;
  recommendations?: {
    decision?: Recommendation | string | null;
    rationale?: string | null;
    nextActions?: string[];
  } | null;
  confidence_level?: ConfidenceLevel | string | null;
  confidence?: {
    level?: ConfidenceLevel | string | null;
    score?: number | null;
    rationale?: string | null;
  };
  summary: string | null;
  strengths: EvidenceItem[];
  gaps?: EvidenceItem[];
  risks?: EvidenceItem[];
  concerns?: EvidenceItem[];
  resume_alignment?: EvaluationSection;
  compensation_alignment?: EvaluationSection;
  seniority_alignment?: EvaluationSection;
  remote_alignment?: EvaluationSection;
  technical_alignment?: EvaluationSection;
  company_risk?: EvaluationSection;
  compensationFindings?: EvaluationSection & {
    notes?: string | null;
    evidence?: string[];
  };
  atsFindings?: {
    matchedKeywords?: string[];
    missingKeywords?: string[];
    keywordNotes?: string | null;
    score?: number | null;
  };
  sections?: {
    strategicFit?: EvaluationSection;
    technicalAlignment?: EvaluationSection;
    seniorityAlignment?: EvaluationSection;
    compensationAlignment?: EvaluationSection;
    remoteAlignment?: EvaluationSection;
    companyRisk?: EvaluationSection;
    applicationEffort?: EvaluationSection;
    atsResumeAlignment?: EvaluationSection;
  };
  assumptions?: string[];
  ats_keywords?: string[];
  missing_keywords?: string[];
  model_used: string | null;
  prompt_version: string | null;
  ruleset_version: string | null;
  input_token_estimate: number | null;
  output_token_estimate: number | null;
  latency_ms: number | null;
  ai_enabled: boolean;
  ai_status: "completed" | "failed" | "skipped" | string | null;
  error_message: string | null;
  role_content_hash: string | null;
  source_hash: string | null;
  evaluation_input_hash: string | null;
  raw_evaluation_json?: {
    ai_status?: "completed" | "failed" | "skipped";
    ai_failure_reason?: string;
    ai_error_type?: string;
    ai_result?: {
      unsupported_claim_warnings?: EvidenceItem[];
      evidence_matches?: EvidenceItem[];
      evidence_gaps?: EvidenceItem[];
      positioning_opportunities?: EvidenceItem[];
      [key: string]: unknown;
    };
    validationIssues?: unknown[];
    active_resume_source?: {
      source_id: string;
      source_name: string;
      source_type: string;
      version_id: string;
      version_label: string;
    } | null;
    [key: string]: unknown;
  };
  created_at: string;
  updated_at: string;
}

export interface CompassEvaluationCreatePayload {
  user_notes?: string | null;
  user_context?: Record<string, unknown>;
  force?: boolean;
}

export type EvaluationSummaryState =
  | { status: "loading" }
  | { status: "not_evaluated" }
  | { status: "error"; message: string }
  | { status: "completed"; evaluation: CompassEvaluation };

File: backend/app/schemas/compass_evaluations.py
import uuid
from datetime import datetime
from decimal import Decimal
from typing import Any

from pydantic import BaseModel, ConfigDict, Field

from app.constants import CompassEvaluationStatus


class CompassEvaluationCreate(BaseModel):
    user_notes: str | None = Field(default=None, max_length=5000)
    user_context: dict[str, Any] = Field(default_factory=dict)
    force: bool = False


class CompassEvaluationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    workspace_id: uuid.UUID
    role_id: uuid.UUID
    evaluation_status: CompassEvaluationStatus
    overall_score: Decimal | None
    recommendation: str | None
    confidence_level: str | None
    summary: str | None
    strengths: list[dict[str, Any]]
    concerns: list[dict[str, Any]]
    resume_alignment: dict[str, Any]
    compensation_alignment: dict[str, Any]
    seniority_alignment: dict[str, Any]
    remote_alignment: dict[str, Any]
    technical_alignment: dict[str, Any]
    company_risk: dict[str, Any]
    ats_keywords: list[str]
    missing_keywords: list[str]
    model_used: str | None
    prompt_version: str | None
    ruleset_version: str | None
    input_token_estimate: int | None
    output_token_estimate: int | None
    latency_ms: int | None
    ai_enabled: bool
    ai_status: str | None
    error_message: str | None
    role_content_hash: str | None
    source_hash: str | None
    evaluation_input_hash: str | None
    raw_evaluation_json: dict[str, Any]
    created_at: datetime
    updated_at: datetime
